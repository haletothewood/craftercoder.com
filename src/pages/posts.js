import React from "react"

import { Helmet } from "react-helmet"
import { Link, graphql } from "gatsby"

import Layout from "../components/layout"

import "../styles/tag-listing.scss"

export const PurePosts = ({ data }) => {
  const { edges: posts } = data.allMarkdownRemark
  const { title } = data.site.siteMetadata
  return (
    <>
      <Helmet title={title} />
      <div>
        <h1>Posts</h1>
        <ul>
          {posts.map(post => (
            <li key={post.node.frontmatter.title}>
              <Link className="tag-listing" to={post.node.frontmatter.path}>
                {post.node.frontmatter.title}
              </Link>
            </li>
          ))}
        </ul>
      </div>
    </>
  )
}

export const Posts = props => {
  return (
    <Layout>
      <PurePosts {...props} />
    </Layout>
  )
}

export default Posts

export const pageQuery = graphql`
  query PostsQuery {
    site {
      siteMetadata {
        title
      }
    }
    allMarkdownRemark(sort: { order: DESC, fields: [frontmatter___date] }) {
      edges {
        node {
          frontmatter {
            title
            date(formatString: "MMMM DD, YYYY")
            path
          }
        }
      }
    }
  }
`
