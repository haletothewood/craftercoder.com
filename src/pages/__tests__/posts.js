import React from "react"
import renderer from "react-test-renderer"
import { PurePosts as Posts } from "../posts"

const props = {
  data: {
    allMarkdownRemark: {
      edges: [
        {
          node: {
            frontmatter: {
              title: "Title",
              path: "/path",
              date: "01/01/2019",
            },
          },
        },
      ],
    },
    site: {
      siteMetadata: {
        title: "Crafter Coder",
      },
    },
  },
}

describe("Posts", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<Posts {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
