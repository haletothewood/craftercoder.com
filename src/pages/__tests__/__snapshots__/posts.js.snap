// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Posts renders correctly 1`] = `
<div>
  <h1>
    Posts
  </h1>
  <ul>
    <li>
      <a
        className="tag-listing"
        href="/path"
      >
        Title
      </a>
    </li>
  </ul>
</div>
`;
